File: alexandria/ui/src/io/intino/alexandria/ui/displays/components/Item.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.Component;
import io.intino.alexandria.ui.displays.notifiers.ItemNotifier;

public class Item<DN extends ItemNotifier, Type, B extends Box> extends Component<DN, B> {
    private Type item;

    public Item(B box) {
        super(box);
    }

    public Type item() {
        return this.item;
    }

    public Item item(Type item) {
        this.item = item;
        return this;
    }

}

File: web-ui/src/displays/components/Item.js
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import AbstractItem from "../../../gen/displays/components/AbstractItem";
import ItemNotifier from "../../../gen/displays/notifiers/ItemNotifier";
import ItemRequester from "../../../gen/displays/requesters/ItemRequester";
import DisplayFactory from 'alexandria-ui-elements/src/displays/DisplayFactory';

const styles = theme => ({});

class Item extends AbstractItem {

	constructor(props) {
		super(props);
		this.notifier = new ItemNotifier(this);
		this.requester = new ItemRequester(this);
	};


}

export default withStyles(styles, { withTheme: true })(Item);
DisplayFactory.register("Item", withStyles(styles, { withTheme: true })(Item));